lib/IRGen/GenClangType.cpp

```diff
@@ -585,8 +585,10 @@ clang::CanQualType GenClangType::visitSILFunctionType(CanSILFunctionType type) {
   if (allResults.empty()) {
     resultType = clangCtx.VoidTy;
   } else {
-    resultType = Converter.convert(IGM,
-                   allResults[0].getReturnValueType(IGM.getSILModule(), type));
+    resultType = Converter.convert(
+        IGM,
+        allResults[0].getReturnValueType(IGM.getSILModule(), type,
+                                         IGM.getMaximalTypeExpansionContext()));
```

Comment:
Could you update the code in `ClangTypeConverter.cpp` to match the changes in this file? I've written a more detailed explanation here: https://github.com/apple/swift/pull/30630#discussion_r405057467 for why we have this duplication at the moment.

Reply:
There is no matching change to this. `some P` types where this would be relevant should not occur in Objective-C types.